src/main/scala/example/rest/UserRoutes.scala

```diff
@@ -0,0 +1,39 @@
+package example.rest
+
+import akka.http.scaladsl.server.Directives._
+import example.domain.{NewUser, User}
+import example.service.UserServiceFuture
+import example.utils.Retrier
+import scala.concurrent.duration._
+
+/**
+  * Controller for routing between application endpoints.
+  * @param userService injected User Service to use business logic operations from.
+  */
+class UserRoutes(userService: UserServiceFuture) extends JsonSupport {
+
+  val routes =
+    pathPrefix("users") {
+      post {
+        entity(as[NewUser]) { user =>
+          complete(
+            new Retrier().retry(
+              () => userService.registerUser(user.username, user.address, user.email),
+              (a: Either[String, User]) => a.isRight,
```

Comment:
oh, i see, you added retries on that level but that is wrong. This will retry all the validation errors which doesn't make sense.